merino/src/docs/api.rs

```diff
@@ -143,4 +153,31 @@ the user opts out the telemetry collection of Firefox.
 
 The required behavior for interaction pings is TBD.
 
+## Providers
+
+Endpoint: `/api/v1/providers`
+
+This endpoint gives a list of available providers, along with their
+*availability*.  It accepts GET requests and takes no parameters.
+
+### Response
+
+The response will be a JSON object containing the key `providers`, which be a
+map where the keys to this map are the IDs of the provider, and the value are
+provider metadata object. Each provider metadata object will have the following
+format:
+
+- `id` - A string that can be used to identify this provider. This ID can be
+  used for the `providers` field of the suggest API.
+
+- `availability` - A string describing how this provider is used in Merino. It will be one of:
+
+  - `"enabled-by-default"` - This provider will be used for requests that don't
```

Comment:
`ProviderAvailability` is serialized as `snake_case` but described as `kebab-case` here (which I just learned serde does support if you want it 🥙)

Reply:
Oops. Probably better to keep it all `snake_case` for consistency.